### PR TITLE
Redesign the Handlebars demo app (Web, React, Form Validations) #DS-1479

### PR DESCRIPTION
**Pull request by adamkudrna**
## Description

Include header and footer on all demo pages for easier navigation.

### Additional context

obrazek

### Issue reference

https://jira.almacareer.tech/browse/DS-1479

### CODE REVIEW

**Comment by crishpeen** [comment]
Very nice, looking forward 😅😅😅

**Comment by literat** [comment]
Do not forget to update the snapshot tests in web-twig.

**Comment by adamkudrna** [comment]
> 
> 
> Very nice, looking forward 😅😅😅

I reduced the scope of this PR to the Handlebars demo only, without the theme switcher. More PRs are on their way! 🎉 

obrazek

**Comment by adamkudrna** [comment]
Split into #1675 (Twig), #1676 (theme switcher), #1677 (visual tests update).

**Review by literat — APPROVED** [review]
Nice work. I am looking forward to some future demo updates. 👍

File: apps/demo/partials/cover.hbs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      style="--flex-spacing: var(--spirit-space-1000)"
+    >
+      {{title}}
+
+      {{#if isUnstable}}
+        <span class="Tag Tag--large Tag--warning">Unstable</span>
+      {{/if}}
+    </h1>
+
+    <nav aria-label="Breadcrumb" class="Breadcrumbs">
```
Comment (crishpeen) [inline apps/demo/partials/cover.hbs:15]:
```suggestion
    <nav aria-label="Breadcrumbs" class="Breadcrumbs">
```
? 🤷

File: apps/demo/partials/cover.hbs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      <ol>
+        <li class="d-flex">
+          <a href="/" class="link-primary link-underlined">Spirit</a>
+          <svg width="24" height="24">
```
Comment (crishpeen) [inline apps/demo/partials/cover.hbs:19]: aria-hidden?

File: apps/demo/partials/header.hbs [hunk 3]
```diff
@@ -1,20 +1,136 @@
...
+              class="HeaderDialogLink {{# if (eq package 'Web React')}}HeaderDialogLink--current{{/if}}"
+              {{# if (eq package 'Web React')}}aria-current="page"{{/if}}
+            >
+              Web React
+            </a>
+          </li>
+          {{!-- Hide link to the Twig implementation in production. Twig demo app only runs on localhost, we don't deploy it anywhere. --}}
+          {{#if @is_development}}
+            <li class="HeaderDialogNavItem">
+              <a href="https://localhost:4443" class="HeaderDialogLink">Web Twig</a>
```
Comment (crishpeen) [inline apps/demo/partials/header.hbs:115]: I would love to see this imported from the common package. Maybe we can create an issue to let someone do it :)

File: apps/demo/partials/web/icons.hbs [hunk 4]
```diff
@@ -1,20 +1,16 @@
...
-        </li>
-      {{/each}}
-
-    {{/tileList}}
-
-  {{/section}}
+  <ul class="Grid Grid--cols-2 Grid--tablet--cols-4 Grid--desktop--cols-6 text-center my-1300">
+    {{#each @icons as |icon|}}
+      <li class="mb-700">
+        <svg width="24" height="24">
```
Comment (crishpeen) [inline apps/demo/partials/web/icons.hbs:8]: aria-hidden?

File: packages/demo/scss/index.scss [hunk 5]
```diff
@@ -1,36 +1,65 @@
...
+    padding: var(--spirit-space-700);
+    border: var(--spirit-border-width-100) solid var(--spirit-color-border-interactive-default);
+    border-radius: var(--spirit-radius-300);
+    background-color: var(--spirit-color-background-primary);
+    font-size: var(--spirit-typography-body-large-semibold-mobile-font-size);
+    font-weight: var(--spirit-typography-body-large-semibold-mobile-font-weight);
+    line-height: var(--spirit-typography-body-large-semibold-mobile-line-height);
+    transition: box-shadow 125ms;
+
+    &:hover {
```
Comment (crishpeen) [inline packages/demo/scss/index.scss:25]: media hover? :))

File: apps/demo/partials/header.hbs (hunk 3 above)
Comment (adamkudrna) [reply apps/demo/partials/header.hbs:115]: Unfortunately, the Vite config refuses to load any data from the `common` package. Maybe @literat will know, but it looks like a separate issue indeed.

```js
// apps/demo/config/vite/app.ts

import { SERVERS } from '@lmc-eu/spirit-common/constants/servers';

export default defineConfig({
  // …
  server: {
    hmr: {
      overlay: false,
    },
    ...SERVERS.DEVELOPMENT.web,
  },
});
```
```
failed to load config from /Users/adam.kudrna/Web/spirit-design-system/apps/demo/config/vite/app.ts
error when starting dev server:
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /Users/adam.kudrna/Web/spirit-design-system/packages/common/constants/environments.ts
```

Comment (adamkudrna) [reply apps/demo/partials/header.hbs:115]: Filed as https://jira.almacareer.tech/browse/DS-1506.

**Review by crishpeen — APPROVED** [review]

**Pull request merged**